app/controllers/wallets_controller.rb

```diff
@@ -0,0 +1,28 @@
+class WalletsController < ApplicationController
+  include WalletFunc
+
+  def new
+    @wallet = Wallet.new
+    @currencies = currencies
+  end
+
+  def create
+    @wallet = current_user.wallets.new(currency_id: wallet_params[:currency_id], amount: 0,
+                                       wallet_number: create_random_wallet_number,
+                                       user_id: current_user.id, locked: false)
+
+    respond_to do |format|
+      if @wallet.save
+        format.html { redirect_to current_user, notice: 'Wallet was successfully created.' }
+      else
+        format.html { render :new, location: @wallet }
+      end
```

Comment:
we agreed that it makes no sense and we will not add it

Reply:
Liza had to redo it when she added a pop-up, I'll fix it